internal/rbd/controllerserver.go

```diff
@@ -963,7 +972,7 @@ func (cs *ControllerServer) ValidateVolumeCapabilities(
 }
 
 // CreateSnapshot creates the snapshot in backend and stores metadata in store.
-// nolint:cyclop // TODO: reduce complexity
+// nolint:gocyclo,cyclop // TODO: reduce complexity.
```

Comment:
adding nolint is not helpful we will forget about it once PR is merged and no use of having linter checks in CI isn't it?

Reply:
This function is already complex and at its best-simplified level. You can see all it already has `cyclop`.
Excuses apart, we should consider scrubbing such complex functions sometimes.